File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        # min_v=arrays[0][0]
        # max_v = arrays[0][-1]
        # maxi=0
        # for i in range(1,len(arrays)):
        #     cu_min=arrays[i][0]
        #     cu_max=arrays[i][-1]
        #     maxi=max(maxi,abs(min_v - cu_max), abs(max_v - cu_min))
        # min_v=min(min_v,cu_min)
        # mac_v=min(max_v, cu_max)
        # return maxi
            min_value = arrays[0][0]
            max_value = arrays[0][-1]
            max_distance = 0
            
            for i in range(1, len(arrays)):
                current_min = arrays[i][0]
                current_max = arrays[i][-1]
                
                # Calculate the maximum possible distance with the current array
                max_distance = max(max_distance, abs(current_max - min_value), abs(max_value - current_min))
                
                # Update min and max values for the next iteration
                min_value = min(min_value, current_min)
                max_value = max(max_value, current_max)
            
            return max_distance

File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.java
import java.util.List;

class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int min_val = arrays.get(0).get(0); // Access first element of the first list
        int max_val = arrays.get(0).get(arrays.get(0).size() - 1); // Access last element of the first list
        int maxi = 0;

        for (int i = 1; i < arrays.size(); i++) { // Start loop from 1, since we initialized with 0th list
            int c_min = arrays.get(i).get(0); // Access first element of current list
            int c_max = arrays.get(i).get(arrays.get(i).size() - 1); // Access last element of current list
            
            // Calculate the maximum distance considering the current list
            maxi = Math.max(maxi, Math.max(Math.abs(min_val - c_max), Math.abs(max_val - c_min)));

            // Update min_val and max_val with the current list's values
            min_val = Math.min(min_val, c_min);
            max_val = Math.max(max_val, c_max);
        }
        return maxi;
    }
}
